src/core/core.scale.js

```diff
@@ -67,6 +67,25 @@ defaults._set('scale', {
 	}
 });
 
+function sample(arr, size) {
```

Comment:
One reason I hadn't done that is that potentially there's some pattern in the labels, so getting a random sample of labels is likely to lead to better outcomes. E.g. imagine that we're trying to run the auto-skipper on a linear scale. If we take just the first ticks then they'll be small labels with just a few characters whereas if you have large numbers at the other end of the scale like 1,000,000 then we might be underestimating the size of the average label.

Reply:
I dislike the random, because we could get different result between updates.
`sample` above returns ticks 0, 1, 3, 6, 10, 15, 21, ...
Could also do the same from both ends (returning 0, 1, 3, 6, 10, 15, 21, ...  , -21, -15, -10, -6, -3, -1, end):
```js
function sample(arr, size) {
	var ret = [];
	var i, len, p;

	for(i = 0, p = 0, len = arr.length; i < size / 2 && p < len / 2; ++i, p += i) {
		ret[i] = arr[p];
		ret[size - i - 1] = arr[len - p - 1];
	}
	return ret;
}
```

Reply:
https://jsperf.com/tick-sampling/1

Reply:
Ok. I've pushed a new version that removes the randomness